dbt/adapters/sql/connections.py

```diff
@@ -199,3 +211,46 @@ def commit(self):
         connection.transaction_open = False
 
         return connection
+
+    def _retryable_cursor_execute(self,
+                                  execute_fn: Callable,
+                                  sql: str,
+                                  bindings: Optional[Any] = None,
+                                  retryable_exceptions: Iterable[Type[Exception]] = [],
+                                  retry_limit: int = 1,
+                                  retry_timeout: Union[Callable[[int], SleepTime], SleepTime] = 1,
+                                  _attempts: int = 0,
```

Comment:
This feels like an early abstraction. Can we start with a timeout that's just an integer and go from there?